### PR TITLE
Github Action Migration

### PR DESCRIPTION
**Pull request by HarrisonWAffel**
A successful release run can be seen here: https://github.com/HarrisonWAffel/terraform-provider-rancher2/actions/runs/9135094356
The github release and artifacts can be seen here: https://github.com/HarrisonWAffel/terraform-provider-rancher2/releases/tag/v0.0.0

A successful pre-release run can be seen here: https://github.com/HarrisonWAffel/terraform-provider-rancher2/actions/runs/9134965367
The github release and artifacts can be seen here: https://github.com/HarrisonWAffel/terraform-provider-rancher2/releases/tag/v0.0.0-rc1

I've checked the GPG signing process using my personal GPG key 

```
gpg: Signature made Fri May 17 23:01:41 2024 UTC
gpg:                using RSA key B1BB4CAE50A884D616C4DCF69A6B757E291A63DF
gpg: Good signature from "Harrison Affel <harrison.affel@suse.com>" [unknown]
Primary key fingerprint: B1BB 4CAE 50A8 84D6 16C4  DCF6 9A6B 757E 291A 63DF
```

In a similar fashion as the rke terraform provider, I've removed the `golint` step in the make file. `golint` is no longer maintained and does not work properly with go modules. We should introduce a better linting solution in the near future, however this PR currently only focuses on GHA migration.

SURE-8006

### CODE REVIEW

**Review by jakefhyde — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v1
```
Comment (jiaqiluo) [inline .github/workflows/release.yaml]: Q: Any reason for not using v4 in both workflows?

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      - name: package
+        run: |
+          make package-rancher
+
+            - name: retrieve GPG Credentials
+              uses: rancher-eio/read-vault-secrets@main
+              with:
+                secrets: |
+                  secret/data/github/repo/${{ github.repository }}/key/app-credentials passphrase | GPG_PASSPHRASE ;
+                  secret/data/github/repo/${{ github.repository }}/key/app-credentials privateKey | GPG_KEY
```
Comment (jiaqiluo) [inline .github/workflows/release.yaml]: Can you fix the indents?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (HarrisonWAffel) [reply .github/workflows/release.yaml]: I've upgraded to v4

File: .github/workflows/release.yaml (hunk 2 above)
Comment (HarrisonWAffel) [reply .github/workflows/release.yaml]: Good catch, copy paste error on my part, should be fixed

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**